### PR TITLE
enhancement/mui-header

### PR DESCRIPTION
**Pull request by Markusplay**

### CODE REVIEW

File: src/components/common/layout/header/Header.tsx [hunk 1]
```diff
@@ -1,36 +1,10 @@
...
-
-import { HeaderDesktopCard } from './components/header-desktop-card';
-import HeaderDivider from './components/header-divider/HeaderDivider';
-import { HeaderMobileCard } from './components/header-mobile-card/HeaderMobileCard';
-
-import styles from './Header.module.scss';
+import DesktopHeader from './components/desktop-header/DesktopHeader';
+import MobileHeader from './components/mobile-header/MobileHeader';
 
 const roleMapper = {
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/Header.tsx:9]: Make another folder called 'utils' and make function transformData. Move roleMapper and all of the user data tranformation there

File: src/components/common/layout/header/Header.tsx [hunk 2]
```diff
@@ -44,281 +18,25 @@ const Header: React.FC = () => {
...
-                variant={ButtonVariant.FILLED}
-              />
-            </Link>
-          </div>
-        )}
-      </div>
-    )
+  const isMobile = useMediaQuery('(max-width:1200px)');
+  return isMobile ? (
+    <MobileHeader
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/Header.tsx:23]: Let MobileHeader and DesktopHeader have 2 props:
1. user (object, which transformData will return)
2. isLoggedIn

File: src/components/common/layout/header/components/mobile-header/MobileHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import { TabItemContentSize } from '@/components/common/ui/tab/tab-item/TabItem';
+
+import * as styles from '../../Header.styles';
+import { DesktopHeaderProps } from '../desktop-header/DesktopHeader';
+import HeaderDivider from '../header-divider';
+import { HeaderMobileCard } from '../header-mobile-card';
+import HeaderMobileMenu from '../header-mobile-menu/HeaderMobileMenu';
+
+interface MobileHeaderProps extends DesktopHeaderProps {
+  isLoggedIn: boolean;
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.tsx:31]: DesktopHeaderProps already has isLoggedIn

File: src/components/common/layout/header/components/header-mobile-menu/HeaderMobileMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import { TabItemContentSize } from '@/components/common/ui/tab/tab-item/TabItem';
+
+import * as styles from '../../Header.styles';
+interface HeaderMobileMenuProps {
+  setIsOpened: any;
+}
+
+const HeaderMobileMenu: React.FC<HeaderMobileMenuProps> = ({ setIsOpened }) => {
+  return (
+    <Box sx={styles.mobileMenu}>
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/header-mobile-menu/HeaderMobileMenu.tsx:21]: What is this menu? Why it's not using constants?

File: src/components/common/layout/header/components/mobile-header/MobileHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                <Button
+                  text="Увійти"
+                  size={ButtonSize.SMALL}
+                  variant={ButtonVariant.FILLED}
+                />
+              </Link>
+            </Box>
+          </Box>
+          <HeaderDivider />
+          <HeaderMobileMenu setIsOpened={handleClick} />
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.tsx:162]: Duplicate <HeaderMobileMenu>. Same as code row number 99-100

File: src/components/common/layout/header/components/mobile-header/MobileHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+        </Box>
+      </Box>
+    ) : (
+      <Box sx={styles.headerContainer}>
+        <Link href={'/'}>
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.tsx:167]: Too much nesting. At least use Box to display image, and if possible put href on this box too

File: src/components/common/layout/header/components/mobile-header/MobileHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+            <CloseButton
+              onClick={handleClick}
+              size={IconButtonSize.MEDIUM}
+              color={IconButtonColor.TRANSPARENT}
+            />
+          </Box>
+        </Box>
+        <Box sx={styles.drop}>
+          <Box sx={styles.loginButtons}>
+            <Box style={{ width: '192px' }}>
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.tsx:142]: inline styles

File: src/components/common/layout/header/components/mobile-header/MobileHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+        </Box>
+      </Box>
+    );
+  }
+
+  if (!isLoggedIn) {
+    return isOpened ? (
+      <Box>
+        <Box sx={styles.shadow} onClick={handleClick}></Box>
+        <Box sx={styles.headerContainer} style={{ backgroundColor: '#1e1e1e' }}>
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.tsx:126]: inline styles

File: src/components/common/layout/header/components/mobile-header/MobileHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+  const [isOpened, setIsOpened] = useState(false);
+  const handleClick = () => {
+    setIsOpened(isOpened => !isOpened);
+  };
+
+  if (isLoggedIn) {
+    return isOpened ? (
+      <Box>
+        <Box sx={styles.shadow} onClick={handleClick} />
+        <Box sx={styles.headerContainer} style={{ backgroundColor: '#1e1e1e' }}>
```
Comment (OleksandrHumeniuk) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.tsx:50]: inline styles

**Review by OleksandrHumeniuk — CHANGES_REQUESTED** [review]
Also:
Rewrite desktop and mobile cards as mui components. If possible maybe even make it one component with desktop/mobile prop. For user profile picture use mui Avatar component.

File: src/components/common/layout/header/Header.tsx (hunk 1 above)
Comment (Markusplay) [reply src/components/common/layout/header/Header.tsx:9]: If we squash mobile and desktop headers into one component, utils don`t needed

**Review by OleksandrHumeniuk — APPROVED** [review]

**Review by OleksandrHumeniuk — CHANGES_REQUESTED** [review]
Split Header into MobileHeader and DesktopHeader into 2 and add dynamic important like so:

image

**Review by OleksandrHumeniuk — APPROVED** [review]

**Pull request merged**